ni_measurementlink_service/measurement/service.py

```diff
@@ -181,10 +181,18 @@ def __init__(
             ui_file_paths=ui_file_paths,
         )
 
+        service_annotations_string: Dict[str, str] = {}
+        if "annotations" in service:
+            for keys in service["annotations"]:
+                service_annotations_string[keys] = json.dumps(
+                    service["annotations"][keys], separators=(",", ":")
+                )
```

Comment:
Simplify using get(), items(), and dict comprehensions.

The loop variable should be singular.

You probably don't need a type hint here because `json.dumps` returns a `str`: https://github.com/python/typeshed/blob/main/stdlib/json/__init__.pyi#L13

```suggestion
        service_annotations_string = {
            key: json.dumps(value, separators=(",", ":")
            for key, value in service.get("annotations", {}).items()
        }
```